kolibri/plugins/user_profile/assets/src/views/ChangeFacility/SelectFacility.vue

```diff
@@ -0,0 +1,279 @@
+<template>
+
+  <div>
+    <span class="headercontainer">
+      <h1>{{ $tr('documentTitle') }}</h1>
+
+      <transition name="spinner-fade">
+
+        <div v-if="discoveringPeers">
+          <KLabeledIcon>
+            <template #icon>
+              <KCircularLoader :size="16" :stroke="6" class="loader" />
+            </template>
+          </KLabeledIcon>
+        </div>
+      </transition>
+
+    </span>
+    <p v-if="initialFetchingComplete && !availableFacilities.length">
+      {{ $tr('noFacilitiesText') }}
+    </p>
+    <template v-for="f in availableFacilities">
+      <div :key="`div-${f.id}`">
+        <KRadioButton
+          :key="f.id"
+          v-model="selectedFacilityId"
+          class="radio-button"
+          :value="f.id"
+          :label="formatNameAndId(f.name, f.id)"
+          :disabled="discoveryFailed || !isAddressAvailable(f.address_id)"
+        />
+      </div>
+
+    </template>
+
+    <KGrid
+      :style="{
+        marginTop: '34px',
+        paddingTop: '10px',
+        borderTop: `1px solid ${$themePalette.grey.v_300}`
+      }"
+    >
+
+      <KGridItem>{{ $tr('doNotSeeYourFacility') }}</KGridItem>
+      <KGridItem>
+        <KButton
+          class="new-address-button"
+          :text="$tr('newAddressButtonLabel')"
+          appearance="basic-link"
+          @click="showAddAddressModal = true"
+        />
+      </KGridItem>
+    </KGrid>
+
+    <AddAddressForm
+      v-if="showAddAddressModal"
+      @cancel="showAddAddressModal = false"
+      @added_address="handleAddedAddress"
+    />
+    <BottomAppBar>
+      <slot name="buttons">
+        <KButtonGroup>
+          <KButton
+            :primary="true"
+            :text="coreString('continueAction')"
+            :disabled="selectedFacilityId === ''"
+            @click="to_continue"
+          />
+        </KButtonGroup>
+      </slot>
+    </BottomAppBar>
+
+  </div>
+
+</template>
+
+
+<script>
+
+  import { useLocalStorage } from '@vueuse/core';
+  import { computed } from 'kolibri.lib.vueCompositionApi';
+  import responsiveWindowMixin from 'kolibri.coreVue.mixins.responsiveWindowMixin';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import commonSyncElements from 'kolibri.coreVue.mixins.commonSyncElements';
+  import client from 'kolibri.client';
+  import urls from 'kolibri.urls';
+  import BottomAppBar from 'kolibri.coreVue.components.BottomAppBar';
+  import useSavedAddresses from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/useSavedAddresses.js';
+  import useDynamicAddresses from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/useDynamicAddresses.js';
+  import AddAddressForm from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/AddAddressForm';
+
+  export default {
```

Comment:
When creating completely new components that use Composable API, I don't see a reason why to mix it with the Options API even though it's technically compatible. I think this strategy is useful when introducing composables to existing components because it doesn't require refactoring a whole component, however for new components I'd say it's a bit confusing since you don't know if you should go to `setup` or to the options. I know about one pitfall with accessing translations in `setup` so I think we may need to use the Options API for properties that need to access `$tr` but other than that, were there any other reasons for mixing both APIs?

Reply:
sorry, I don't understand this comment. The api I'm using there is not created in this PR, it's an existing api that already exists in Kolibri and it's used in other components. The components are new and they're using these functions in the `setup` section. What are you proposing exactly?

Reply:
I meant Vue APIs
- Composition API - using `setup` and defining methods, computed properties, watchers etc. inside of `setup`
- Options API - declared options (methods, computed, watchers etc.), defined as properties on the object being exported

For example, in this component, `combinedAddresses` is implemented as a computed property by means of the Composition API whereas `isAddressAvailable` is implemented as a computed property by means of the Options API.

I'm proposing using one consistent style when creating new components instead of mixing these two.

Reply:
No need to refactor this one, but would be nice to keep this in mind whenever we know that we need to use the Composition API in a new component. Then we can implement it fully using Composition API so that in the future there's no need to spend time locating methods, watchers, and all the things in two places.